src/output_queue/output_queue.py

```diff
@@ -1,16 +1,101 @@
+from ctypes import c_bool
 import mido
+import time
+from threading import Thread
 
-class OutputQueue:
-    # Test fucntions for setting up unit test infra. TODO: Remove in sprint 2.
-    def is_even(self, number):
-        if number % 2 == 0:
-            return True
-        return False
-
-    # Test fucntions for setting up unit test infra. TODO: Remove in sprint 2.
-    def in_range(self, number):
-        lower = 3
-        upper = 8
-        if number > lower and number < upper:
-            return True
-        return False
\ No newline at end of file
+from src.common.midi_event import MidiEvent
+
+class OutputQueue(Thread):
+    def __init__(self, queue):
+        Thread.__init__(self)
+        self.queue = queue
+        self._open_port = None
+        self._running = False
+
+    def __del__(self):
+        if self._open_port != None:
+            self._open_port.close()
+
+    # Selects the output device to send MIDI to. If `name` is None then the system default is used
+    def select_device(self, name=None):
+        if not name == None and not name in self.get_device_list():
+            raise Exception('"{}" does not match any of the available devices'.format(name))
+
+        if self._open_port != None:
+            self._open_port.close()
+
+        self._open_port = mido.open_output(name)
+
+        print('Switched output device to "{}"'.format(self._open_port.name))
+
+    # Returns the list of currently available MIDI devices that can be connected to
+    def get_device_list(self):
+        return mido.get_output_names()
+        
+    # Checks the queue for messages and sends them to the output as needed and returns the number of message sent (mainly for testing)
+    def _check_queue(self):
+        if self._open_port == None:
+            return 0
```

Comment:
Can you do -1 as the error code? Otherwise you can't tell if it 0ed due to no messages or a closed port

Reply:
I'm just removing that all together. I did this before I mocked the output port and I can get sent messages from that instead.